### PR TITLE
microsoft365defender:feature - Added 'update incident' action

### PR DESCRIPTION
**Pull request by bliew-splunk**
Please ensure your pull request (PR) adheres to the following guidelines:

- Please refer to our contributing documentation for any questions on submitting a pull request, link: [Contribution Guide](https://github.com/splunk-soar-connectors/.github/blob/main/.github/CONTRIBUTING.md)

#### Please check if your PR fulfills the following requirements:
- [X] Testing of all the changes has been performed (for bug fixes / features)
- [X] The manual_readme_content.md has been reviewed and added / updated if needed (for bug fixes / features)
- [X] Use the following format for the PR description: `<App Name>: <PR Type> - <PR Description>`
- [X] Provide release notes as part of the PR submission which describe high level points about the changes for the upcoming GA release.
- [ ] Verify all checks are passing.
- [X] Do NOT use the `next` branch of the forked repo. Create separate feature branch for raising the PR.
- [ ] Do NOT submit updates to dependencies unless it fixes an issue.
#### Please check the type of change your PR introduces:
- [ ] New App
- [ ] Bugfix
- [X] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Documentation
- [ ] Other (please describe):

## Security Considerations (REQUIRED)
- [ ] If you are exposing any endpoints using a [REST handler](https://docs.splunk.com/Documentation/SOAR/current/DevelopApps/RESTHandlers),
  please document them in the `manual_readme_content.md`.
- [X] If this is a new connector or you are adding new actions
    - [X] Please document in the `manual_readme_content.md` all methods (eg, OAuth) used to authenticate
      with the service that the connector is integrating with.
    - [ ] If any actions are unable to run on SOAR Cloud, please document this in the `manual_readme_content.md`.
- [ ] Are you introducing any new cryptography modules? If yes, please elaborate their purpose:
- [ ] Are you are accessing the file system? If yes, please verify that you are only accessing paths returned through
the [Vault](https://docs.splunk.com/Documentation/SOARonprem/5.2.1/DevelopApps/AppDevAPIRef#Vault) API.
- [ ] Are you are marking code to be ignored by Semgrep with [`nosemgrep`](https://semgrep.dev/docs/ignoring-files-folders-code/#ignoring-code-through-nosemgrep)?
If yes, please provide justification in an additional comment next to the ignored code.

## Release Notes (REQUIRED)
- Added 'update incident' action which corresponds to the MS Graph API endpoint: https://learn.microsoft.com/en-us/graph/api/security-incident-update?view=graph-rest-1.0&tabs=http

## What is the current behavior? (OPTIONAL)
- Describe the current behavior that you are modifying.

## What is the new behavior? (OPTIONAL)
- Describe the behavior or changes that are being added by this PR.

## Other information (OPTIONAL)
- Any other information that is important to this PR such as screenshots of how the component looks before and after the change.

## Pay close attention to (OPTIONAL)
- Any specific code change or test case points which must be addressed/reviewed at the time of GA release.

## Screenshots (if relevant)

---
Thanks for contributing!

### CODE REVIEW

**Comment by cammurray** [comment]
Hey! From MSFT here. Be great to see this one get checked in. Any assistance needed on this, please connect with me.

File: microsoft365defender_consts.py [hunk 1]
```diff
@@ -126,6 +126,8 @@
 DEFENDER_NO_ALERT_FOUND = "No alert found"
 DEFENDER_NO_INCIDENT_FOUND = "No incident found"
 DEFENDER_ALERT_UPDATED_SUCCESSFULLY = "Successfully updated the alert"
+DEFENDER_INCIDENT_UPDATED_SUCCESSFULLY = "Successfully updated the incident"
+DEFENDER_INCIDENT_NO_PARAMETER_PROVIDED = "Please provide at least one of the property to update the incident"
```
Comment (tonyc-phantom) [inline microsoft365defender_consts.py]: "one of the property"? Should it be "one of the properties"?

File: release_notes/unreleased.md [hunk 2]
```diff
@@ -1 +1,2 @@
 **Unreleased**
+* Added 'update incident' action
```
Comment (tonyc-phantom) [inline release_notes/unreleased.md]: I think we should be adding the PAPP ticket to the end of this line

**Review by tonyc-phantom — DISMISSED** [review]

**Comment by tonyc-phantom** [comment]
Static test failure - `update incident` block is not present in a test playbook

**Review by bbielinski-splunk — CHANGES_REQUESTED** [review]
Do you maybe have any screenshots of the added view? It would be really helpful to validate the view. Apart from that and the failed static tests, this PR looks good to me.

File: microsoft365defender_consts.py (hunk 1 above)
Comment (ishans-crest) [reply microsoft365defender_consts.py]: Done

File: release_notes/unreleased.md (hunk 2 above)
Comment (ishans-crest) [reply release_notes/unreleased.md]: Done

**Review by ishans-crest — APPROVED** [review]

**Review by bbielinski-splunk — APPROVED** [review]

**Pull request merged**